R/add_smr_type.R

```diff
@@ -0,0 +1,75 @@
+#' Add smrtype variable based on record ID
+#'
+#' @param recid A vector of record IDs
+#' @param mpat A vector of mpat values
+#'
+#' @return A vector of SMR types
+#' @export
+#'
+#' @family Codes
+#'
+#' @examples
+#' x <- tibble::tibble(
+#'   recid = c("02B", "02B", "02B", "04B", "00B", "AE2", "PIS", "NRS"),
+#'   mpat = c("1", "2", "0", "", "", "", "", "")
+#' )
+#' add_smr_type(x)
+add_smr_type <- function(recid, mpat = NULL) {
+
+  # Situation where some recids are missing
+  if (any(is.na(recid)) & !all(is.na(recid))) {
+    cli::cli_inform(c("i" = "Some values of {.var recid} are {.val NA},
```

Comment:
Should this not be an error?

Would the function be able to continue (in a meaningful way) if there were missing `recid`s? In the SLFs we'd never expect to have a missing recid. I'd be tempted to change the check to: `if (any(is.na(recid)))` and then use `cli::cli_abort`

Reply:
Changed this to use `cli_abort` when any `recid` is NA now